### ISSUE TITLE
Color and icon is not applied

### ISSUE THREAD

**Issue by alextegelid** [fabiospampinato/vscode-terminals#77]
Color and icon for the terminals is not working. Setting `"color": "#00ff00"` or `"icon": "github"` for example doesn't do anything.
I've debugged a bit and I can see the colors and icons being set on the terminals, but it's not shown on the actual terminal icon.

Here you can see the color being included in the creationOptions:
<img width="245" alt="CleanShot 2022-09-01_06 17 05@2x" src="https://user-images.githubusercontent.com/7058721/187830494-4c327578-334d-4464-85c3-faa72e0cb090.png">

My terminals.json:
```
{
  "autorun": true,
  "autokill": true,
  "terminals": [
    {
      "name": "Next",
      "commands": ["npm run dev"],
      "color": "#ff0000"
    },

    {
      "name": "Util",
      "color": "#ff0000"
    }
  ]
}
```

**Comment by StoryDevOffical** [comment]
Hello, 

Same here.

On Windows: only icon works but no color applied (terminal.ansiBlue or #ee3355ff)
On MacOSX: no icon and no color working

**Comment by StoryDevOffical** [comment]
true, same behavior on MAC, now i saw it(icon and color) too before reset to default.

**Comment by Safimmi** [comment]
Same here, both mac and windows

**Comment by alextegelid** [comment]
Indeed it seems to be related to that update. I just tried it on 1.6 and that works fine. Maybe this is a bug in VS Code then...

**Comment by SGirousse** [comment]
> Indeed it seems to be related to that update. I just tried it on 1.69.1 and that works fine. Maybe this is a bug in VS Code then...

I'm on fresh and recent install (1.71.2) : 
```txt
Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19043
Sandboxed: No
```

**Comment by fabiospampinato** [comment]
Maybe it will work again if the extension will set these multiple times 🤔 I'll look into that.

**Comment by fabiospampinato** [comment]
It looks like there are no APIs for changing the icon or the color of a terminal after it has been created, so I guess this is just broken in vscode for now.

/cc @Tyriar if you'd like to take a look at that.

**Comment by Tyriar** [comment]
@fabiospampinato do you mean you have an existing `Terminal` object and you want to change the icon and color? This would be a new API request.

**Comment by fabiospampinato** [comment]
@Tyriar Yes-ish, basically this extension is setting the icon and the color when creating each terminal, and it was working fine in the past, but since some recent release apparently it stopped working, and people are [suggesting](https://github.com/fabiospampinato/vscode-terminals/issues/77#issuecomment-1247847001) it's because some new setting is interfering with that, basically overriding what the extension is doing.

So in some sense we don't really need new APIs, those settings that interfere should just be implemented in a way that they don't do anything if the terminal has already a custom icon/color set, which seems probably what should happen.

If that's not implementable for some reason then the only way to restore that functionality for this extension seems to be able to override icon and color again via some new API.

**Comment by Tyriar** [comment]
> basically this extension is setting the icon and the color when creating each terminal, and it was working fine in the past, but since some recent release apparently it stopped working

Can you point to where it's getting set? It you create the terminal it should work, if you are trying to set `Terminal.creationOptions` it should not.

**Comment by fabiospampinato** [comment]
@Tyriar icon and color are being passed to createTerminal: https://github.com/fabiospampinato/vscode-terminals/blob/dbc7080f9c619e323e36f4487ceca5dce4fd068b/src/run.ts

**Comment by Tyriar** [comment]
Thanks, let's track in https://github.com/microsoft/vscode/issues/167643

**Comment by tiageta** [comment]
As a temporary fix, I got it to work by manually assigning the shellPath, even if it is my default one. I've only tested in Windows, though.

```json
"shellPath": "C:\\Program Files\\PowerShell\\7\\pwsh.exe"
```

**Comment by StoryDevOffical** [comment]
On windows, there is no problem anymore on my side, but on MAC the problem still is.

**Comment by sullivan-ben** [comment]
Can confirm adding shellPath fixes the icons on wsl2/ubuntu too.

My terminals.json:

```json
{
    "autorun": true,
    "autokill": true,
    "terminals": [
        {
            "name": "Server",
            "shellPath": "/usr/bin/zsh",
            "icon": "cloud",
            "open": true,
            "focus": true,
            "commands": [
                "cd /home/my-username/projects/my-project/server",
                "npm run start:debug"
            ]
        },
        {
            "name": "Client",
            "shellPath": "/usr/bin/zsh",
            "icon": "browser",
            "open": true,
            "focus": false,
            "commands": [
                "cd /home/my-username/projects/my-project/client",
                "npm start"
            ]
        }
    ]
}
```

**Comment by edu-hilario** [comment]
> > As a temporary fix, I got it to work by manually assigning the shellPath, even if it is my default one. I've only tested in Windows, though.
> > ```json
> > "shellPath": "C:\\Program Files\\PowerShell\\7\\pwsh.exe"
> > ```
> 
> Worked for me too, but only fixed the icon but not the color:
> 
> ` "shellPath": "C:\\Program Files\\Git\\bin\\bash.exe"` EDIT: fixed the color too, when applied to each terminal setting

This did it for me. Thanks!

**Comment by icodeyou** [comment]
Thank you for this workaround ! 
Working on my mac with the following code: 
```
{
      "name": "my terminal",
      "icon": "watch",
      "color": "terminal.ansiCyan",
      "shellPath": "/bin/zsh",
}
```
